### ISSUE TITLE
qemu machine certificate handling

### ISSUE THREAD

**Issue by st1971** [containers/podman#16868]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

In the corporate environment I work in we use a security tool called zscaler, this acts as a man in the middle re-signing ssl/tls requests with it's certificate.  When using podman machine on a mac the certificates are not being correctly handles which makes it impossible to pull images when using podman remote.

Looking at the current machine init code, if i set SSL_CERT_FILE or SSL_CERT_DIR then the file/s are correctly copied into the qemu VM via ignition and environmental variables are set for these so if I connect via `podman machine ssh`  then I can correctly pull images, however when running `podman -c podman-machine-default pull ...` then if fails with a ssl/tls error.  When performing the latter podman in the vm is launched via systemd which is unaware (by design) of shell environment variables.

A simple solution could be to add the line `EnvironmentFile=-/etc/sysconfig/podman` to the podman unit file then populate `/etc/sysconfig/podman` with the necessary environment variables during init or start-up phase.  Have validate that this approach will work by editing the unit files on a running vm.

This may also be cauing problems with the pass through of other environment variables such as proxy details.

Have checked that the behaviour is the same in the in the current main branch of project

Sorry cannot provide exact version details from mac as away from work at the moment

**Steps to reproduce the issue:**

1. set SSL_CERT_FILE environment var to a cert bundle that includes a zscaler cert

2. podman machine init && podman machine start

3. podman run --rm -it docker.io/library/hello-world

**Describe the results you received:**

Failed to pull container with ssl/tls verification error

**Describe the results you expected:**

Hello world output on the console

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
(paste your output here)
```

**Output of `podman info`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked [the Podman Troubleshooting Guide](https://github.com/containers/podman/blob/main/troubleshooting.md)?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by rhatdan** [comment]
Care to open a PR to make that change?

**Comment by st1971** [comment]
Can do, have some free time over christmas

**Comment by st1971** [comment]
This is already fixed in 4.4.0-dev, my initial testing failed as i didn't realise that the environment variable needs to be set when both running `podman machine init` and `podman machine start`, have validated with on a zscaler protected machine.
